backend/app/services/database.py

```
@@ -199,385 +199,357 @@ def _initialize_database(self):
+                conn.commit()
+                return True
         except Exception as e:
-            logger.error(f"Failed to get user by ID: {str(e)}")
-            raise
-    
-    def get_user_projects(self, user_id: str) -> List[Dict[str, Any]]:
-        """Get all projects accessible by a user."""
+            logger.error(f"Failed to acquire lock: {str(e)}")
+            return False
+
+    def release_node_lock(self, script_id: str, node_id: str, user_id: str) -> bool:
```

Ruff (0.14.14)

[warning] Unused method argument: `script_id`

(ARG002)